config.go

```diff
@@ -98,30 +108,86 @@ func WithInternalServiceDefaults() TLSOption {
 			// then we can drop this.
 			tls.CurveP256,
 		}
+		return nil
 	}
 }
 
 // WithIdentity sets the identity of the server or client which will be
 // presented to its peer upon connection.
 func WithIdentity(cert tls.Certificate) TLSOption {
-	return func(c *tls.Config) {
+	return func(c *tls.Config) error {
+		c.Certificates = []tls.Certificate{cert}
+		return nil
+	}
+}
+
+// WithIdentityFromFile sets the identity of the server or client which will be
+// presented to its peer upon connection from provided cert and key files.
+func WithIdentityFromFile(certPath string, keyPath string) TLSOption {
+	return func(c *tls.Config) error {
+		cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		if err != nil {
+			return fmt.Errorf("failed to load keypair: %s", err.Error())
+		}
 		c.Certificates = []tls.Certificate{cert}
+		return nil
 	}
 }
 
 // WithClientAuthentication makes the server verify that all clients present an
 // identity that can be validated by the certificate pool provided.
 func WithClientAuthentication(authority *x509.CertPool) ServerOption {
-	return func(c *tls.Config) {
+	return func(c *tls.Config) error {
 		c.ClientAuth = tls.RequireAndVerifyClientCert
 		c.ClientCAs = authority
+		return nil
+	}
+}
+
+// WithClientAuthenticationFromFile makes the server verify that all clients present an
+// identity that can be validated by the CA file provided.
+func WithClientAuthenticationFromFile(caPath string) ServerOption {
+	return func(c *tls.Config) error {
+		caBytes, err := ioutil.ReadFile(caPath)
+		if err != nil {
+			return fmt.Errorf("failed read ca cert file: %s", err.Error())
+		}
+
+		caCertPool := x509.NewCertPool()
+		if ok := caCertPool.AppendCertsFromPEM(caBytes); !ok {
+			return fmt.Errorf("Unable to load caCert")
+		}
+
+		c.ClientAuth = tls.RequireAndVerifyClientCert
+		c.ClientCAs = caCertPool
+		return nil
 	}
 }
 
-// WithAuthority makes the server verify that all clients present an identity
+// WithAuthority makes the client verify that the server presents an identity
 // that can be validated by the certificate pool provided.
 func WithAuthority(authority *x509.CertPool) ClientOption {
-	return func(c *tls.Config) {
+	return func(c *tls.Config) error {
 		c.RootCAs = authority
+		return nil
+	}
+}
+
+// WithAuthorityFromFile makes the client verify that the server presents an identity
+// that can be validated by the CA file provided.
+func WithAuthorityFromFile(caPath string) ClientOption {
+	return func(c *tls.Config) error {
+		caBytes, err := ioutil.ReadFile(caPath)
+		if err != nil {
+			return fmt.Errorf("failed read ca cert file: %s", err.Error())
+		}
+
+		caCertPool := x509.NewCertPool()
+		if ok := caCertPool.AppendCertsFromPEM(caBytes); !ok {
+			return fmt.Errorf("Unable to load caCert")
```

Comment:
Nit: "unable..." (lowercase errors)

Can you also include the path of the file which could not be read in the error? I will help when debugging to work out which file is malformed.

Reply:
We agree this error message should be improved, but didn't do so to keep this library backwards compatible with the lib it's replacing (cloudfoundry/cfhttp). Do you have any ideas to get both better error messages and backwards compatibility?

Reply:
And maybe it's ok to make a breaking change to the error messages, with adequate warning to consumers and good documentation.

Reply:
Will this library be wrapped inside cfhttp?

If so, cfhttp can re-write the error messages to maintain backwards compatibility.

If not, I think this is a breaking change to switch to this library in the first place so we can change up the error messages.

In general, relying on the exact error messages is a difficult contract to maintain. What decisions are people making based on the error messages? Maybe we can add `os.IsNotExist` style functions for those instead?

What do you think?

Reply:
We don't have plans to wrap tlsconfig inside cfhttp. We're leaning towards improving the error messages since there will already be breaking changes when switching to this library. It's a bad pattern to assert against error messages.

We will make the requested changes.